bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java

```diff
@@ -81,10 +82,13 @@ public void testInvalidHosts() {
             try {
                 URI uri = new URI(u);
                 nr.setUri(uri);
-                // If we reach this line the only resolved host should be localhost
-                assertEquals(1, nr.numResolvedHosts());
-                uri = nr.getNextUri();
-                assertEquals("127.0.0.1", uri.getHost());
+
+                if (nr.isUnknownHost()) {
+                    assertEquals(uri, nr.getNextUri());
+                } else {
+                    // If we reach this line the only resolved host should be localhost
+                    assertEquals("127.0.0.1", nr.getNextUri().getHost());
```

Comment:
Is it expected that for some invalid hosts we get some valid uri?

Reply:
It looks so.
In `NetResolver::init()` the `host` part (if null then `path`) of provided uri is used to get ip addresses.
For uri like `tcp://10000.122.165.92:3011` the host is null and the path is empty. In this case `InetAddress::getAllByName(null|"")` returns `localhost` address.

Do you think this should be fixed?

Reply:
@678098 fixed

Reply:
@678098 fixing the tests.

Reply:
@678098 done